app_dart/lib/src/model/firestore/task.dart

```diff
@@ -377,3 +382,21 @@ Task taskToDocument(datastore.Task task) {
     ),
   );
 }
+
+/// A [Task], paired with its associated parent [Commit].
+///
+/// The [Task] model object references its parent [Commit] through the
+/// [Task.commitSha] field, but it does not hold a reference to the associated
+/// [Commit] object (just the relational mapping). This class exists for those
+/// times when the caller has loaded the associated commit from firestore
+/// and would like to pass both the task its commit around.
+class FullTask {
```

Comment:
Just a thought and no pressue, one thing to possibly consider is simply using Records rather than defining a full class.

Reply:
I'm new to using records but really like the idea. I updated to use a record instead of FullTask, hopefully it is a bit simpler